From 937d8ca9bf4c50f8a7dc0fbcf9e6ac23b0fbe033 Mon Sep 17 00:00:00 2001
From: Bernd Waibel <waebbl-gentoo@posteo.net>
Date: Mon, 31 Jan 2022 08:17:24 +0100
Subject: [PATCH 2/2] Backport of Use run() instead of Popen() to avoid need
 for communicate

Original patch commit id a65dbc6f8296562a12407a36f4931a80bbb628b7 by
sliptonic <shopinthewoods@gmail.com>

Signed-off-by: Bernd Waibel <waebbl-gentoo@posteo.net>
--- a/src/Mod/Path/PathScripts/PathSanity.py
+++ b/src/Mod/Path/PathScripts/PathSanity.py
@@ -412,8 +412,7 @@ class CommandPathSanity:
             FreeCAD.Console.PrintMessage('asciidoc file written to {}\n'.format(reportraw))
 
         try:
-            result = os.system('asciidoctor {} -o {}'.format(reportraw,
-                reporthtml))
+            result = subprocess.run(["asciidoctor", reportraw, "-o", reporthtml])
             if str(result) == "32512":
                 msg = "asciidoctor not found. html cannot be generated."
                 QtGui.QMessageBox.information(None, "Path Sanity", msg)
-- 
2.35.0

